### PR TITLE
fix: Avoid string concat and regex inefficiency

### PR DESCRIPTION
**Pull request by rafvasq**
This PR makes the following changes:

- Execute external commands using an array of strings rather than a single string as per https://github.com/kserve/modelmesh/security/code-scanning/2 
- Make the regular expression pattern a bit more efficient
https://github.com/kserve/modelmesh/security/code-scanning/1

### CODE REVIEW

File: src/test/java/com/ibm/watson/modelmesh/ModelMeshTearDownTest.java [hunk 1]
```diff
@@ -315,7 +315,7 @@ public static int getPID(Process process) throws Exception {
     }
 
     public static int killProcess(Process process) throws Exception {
-        int pid = getPID(process);
-        return Runtime.getRuntime().exec("kill -9 " + pid).waitFor();
+        String pid = Integer.toString(getPID(process));
```
Comment (ckadner) [inline src/test/java/com/ibm/watson/modelmesh/ModelMeshTearDownTest.java:318]: this change should be fine although `pid` cannot be entered (or manipulated) by a malicious user (and it's test only code)

File: src/main/java/com/ibm/watson/modelmesh/ModelMesh.java [hunk 2]
```diff
@@ -927,7 +927,7 @@ protected final TProcessor initialize() throws Exception {
     // "type" or "type:p1=v1;p2=v2;...;pn=vn"
     private static final Pattern METRICS_CONFIG_PATT = Pattern.compile("([a-z;]+)(:\\w+=[^;]+(?:;\\w+=[^;]+)*)?");
     // "metric_name" or "metric:name;l1=v1,l2=v2,...,ln=vn,"
-    private static final Pattern CUSTOM_METRIC_CONFIG_PATT = Pattern.compile("([a-z_:]+);(\\w+=[^;]+(?:;\\w+=[^,]+)*)?");
+    private static final Pattern CUSTOM_METRIC_CONFIG_PATT = Pattern.compile("([a-z_:]+);(\\w+=[^;]+(?:;\\w+=[^,]++)*)?");
```
[the remark this answers is not in the document]
Comment (rafvasq) [reply src/main/java/com/ibm/watson/modelmesh/ModelMesh.java:930]: It's specifically tested [here](https://github.com/kserve/modelmesh/blob/main/src/test/java/com/ibm/watson/modelmesh/ModelMeshMetricsTest.java#L216-L217) and largely based off of the regex a couple of lines above it.

**Review by njhill — APPROVED** [review]
Thanks @rafvasq. Performance is not important in these contexts (readability should be the main thing). But the changes are fine with me if they suppress the warnings.

**Review by ckadner — APPROVED** [review]
Thanks @rafvasq 

/lgtm

**Pull request merged**